### PR TITLE
refactor: API agent spinup

### PR DESCRIPTION
**Pull request by KulkarniShashank**
### What ###
Refactoring the code related to API agent spinup, To enhance its efficiency, readability, and maintainability. This includes restructuring the initialization process and optimizing relevant components for improved performance.

### Why ###
The refactoring is motivated by the need to address existing code complexities, reduce redundancy, and ensure a more streamlined and coherent API agent spinup process. 

### How ###

- Code Restructuring:
Reorganize the existing codebase to create a more modular and organized structure.
Identify and eliminate any redundant or unnecessary code blocks.

- Performance Optimization:
Evaluate and enhance the efficiency of API agent initialization routines.
Optimize resource utilization for a more responsive and scalable.

### CODE REVIEW

File: apps/agent-service/src/agent-service.service.ts [hunk 1]
```diff
@@ -160,33 +169,52 @@ export class AgentServiceService {
...
 
       if (AgentSpinUpStatus.PROCESSED === getOrgAgent?.agentSpinUpStatus) {
-        throw new BadRequestException('Your wallet is already processing.');
+        throw new BadRequestException(
+          ResponseMessages.agent.error.walletAlreadyProcessing,
+          { cause: new Error(), description: ResponseMessages.errorMessages.badRequest }
+        );
       }
 
+      if (AgentSpinUpStatus.COMPLETED === getOrgAgent?.agentSpinUpStatus) {
```
Comment (vivekayanworks) [inline apps/agent-service/src/agent-service.service.ts:212]: Can we use Switch case over here?

File: libs/common/src/response-messages/index.ts [hunk 2]
```diff
@@ -160,6 +160,15 @@ export const ResponseMessages = {
             notAbleToSpinUpAgent: 'Agent not able to spin-up',
             alreadySpinUp: 'Agent already spin-up',
             agentUrl: 'Agent url not exist',
+            seedChar: 'seed must be at most 32 characters.',
+            validWalletName: 'Please enter valid wallet name, It allows only alphanumeric values',
```
Comment (vivekayanworks) [inline libs/common/src/response-messages/index.ts:164]: Replace "," with "."

File: libs/common/src/response-messages/index.ts [hunk 3]
```diff
@@ -160,6 +160,15 @@ export const ResponseMessages = {
...
+            platformConfiguration: 'Platform configuration is missing or invalid',
+            apiEndpoint: 'API endpoint is missing in the platform configuration',
+            externalIp: 'External IP is missing in the platform configuration',
+            stringExternalIp: 'External IP must be a string',
+            agentProcess: 'Agent process is invalid or not in a completed state',
+            notAbleToSpinup: 'Agent not able to spin-up',
```
Comment (vivekayanworks) [inline libs/common/src/response-messages/index.ts:170]: Remove "-"

File: apps/agent-service/src/agent-service.service.ts (hunk 1 above)
Comment (KulkarniShashank) [reply apps/agent-service/src/agent-service.service.ts:212]: Using a switch statement is preferable when dealing with multiple 'if-else' cases. If there is only a single 'if' condition, then using an 'if' statement is appropriate. Can I proceed with this approach ?

**Pull request merged**